src/dmd/astbase.d

```diff
@@ -4383,7 +4384,13 @@ struct ASTBase
         {
             /* 'Normalize' the value of the integer to be in range of the type
              */
-            switch (type.toBasetype().ty)
+            const ty = type.toBasetype().ty;
+            if (literalFlag && ty != type.ty)
+            {
+                error("cannot modify an uncontextualized static IntegerExp");
```

Comment:
That is a horrible error message if someone actually manages to trip it. Maybe make it an ICE?